components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/template/APITemplateBuilderImpl.java

```diff
@@ -41,65 +49,100 @@
+	private static final String VELOCITY_TEMPLATE_SYNAPSE_CONFIG_NON_VERSIONED_WEBAPP =
+			"velocity-template_synapse-config_non-versioned-webapp.xml";
+	private static final String VELOCITY_TEMPLATE_SYNAPSE_CONFIG_VERSIONED_WEBAPP =
+			"velocity-template_synapse-config_versioned-webapp.xml";
 
-    private WebApp api;
-    private List<HandlerConfig> handlers = new ArrayList<HandlerConfig>();
+	private WebApp api;
+	private String velocityLoggerName = null;
+	private List<HandlerConfig> handlers = new ArrayList<HandlerConfig>();
 
-    public APITemplateBuilderImpl(WebApp api) {
-        this.api = api;
-    }
+	public APITemplateBuilderImpl(WebApp api) {
+		this.api = api;
+		this.velocityLoggerName = getVelocityLoggerName();
+	}
 
 	@Override
-	public String getConfigStringForTemplate(Environment environment) throws APITemplateException {
-		StringWriter writer = new StringWriter();
-
+	public String getConfigStringForVersionedWebAppTemplate(Environment environment)
+			throws APITemplateException {
+
+		// build the context for template and apply the necessary decorators
+		ConfigContext configcontext = new APIConfigContext(this.api);
+		configcontext = new TransportConfigContext(configcontext, api);
+		configcontext = new ResourceConfigContext(configcontext, api);
+		configcontext = new EndpointURIConfigContext(configcontext, api);
+		configcontext = new SecurityConfigContext(configcontext, api);
+		configcontext = new JwtConfigContext(configcontext);
+		configcontext = new ResponseCacheConfigContext(configcontext, api);
+		configcontext = new HandlerConfigContex(configcontext, handlers);
+		configcontext = new EnvironmentConfigContext(configcontext, environment);
+		configcontext = new TemplateUtilContext(configcontext);
+
+		// @todo: this validation might be better to do when the builder is initialized.
 		try {
-			// build the context for template and apply the necessary decorators
-			ConfigContext configcontext = new APIConfigContext(this.api);
-
-			configcontext = new TransportConfigContext(configcontext, api);
-			configcontext = new ResourceConfigContext(configcontext, api);
-			// this should be initialised before endpoint config context.
-			// configcontext = new EndpointBckConfigContext(configcontext,api);
-			// configcontext = new EndpointConfigContext(configcontext, api);
-			configcontext = new EndpointURIConfigContext(configcontext, api);
-			configcontext = new SecurityConfigContext(configcontext, api);
-			configcontext = new JwtConfigContext(configcontext);
-			configcontext = new ResponseCacheConfigContext(configcontext, api);
-			configcontext = new HandlerConfigContex(configcontext, handlers);
-			configcontext = new EnvironmentConfigContext(configcontext, environment);
-			configcontext = new TemplateUtilContext(configcontext);
-
-			// @todo: this validation might be better to do when the builder is
-			// initialized.
 			configcontext.validate();
+		} catch (Exception e) {
```

Comment:
fixed in commit 3ab7b77e4d4f71cb5b9a9749bc61ad08da788894 of PR https://github.com/wso2/carbon-appmgt/pull/153